### PR TITLE
Add scale_value transformation

### PR DESCRIPTION
**Pull request by quentinmit**
**Description:** <Describe what has changed.>

Adds a "scale_value" transformation that can be used to change values' units or otherwise scale them.

**Link to tracking Issue:** <Issue number if applicable>
None

**Testing:** <Describe what testing was performed and which tests were added.>
Unit tests and example config included as part of the PR

**Documentation:** <Describe the documentation added.>
Example usage documented in the README.md

### CODE REVIEW

**Review by punya — APPROVED** [review]
The first line of the README should also be changed to reflect the new capability.

File: processor/metricstransformprocessor/operation_scale_value.go
Comment (punya) [inline processor/metricstransformprocessor/operation_scale_value.go:1]: Nit: Newly added files should use 2021 as the copyright year.

**Comment by quentinmit** [comment]
This PR is still active and is awaiting review.

**Review by djaglowski — APPROVED** [review]
LGTM (Please fix lint though)

**Comment by quentinmit** [comment]
I think I fixed the lint error and comments.

It sounds like you might prefer naming this `experimental_`, so I added a separate commit to rename this if that would make it more likely to merge this.

**Comment by quentinmit** [comment]
Still not stale.

**Review by tigrannajaryan — DISMISSED** [review]
The functionality looks useful but we need to decide where it lives. It intersects with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2722

https://github.com/open-telemetry/opentelemetry-collector/issues/3185 is an open issue that aims to find a good overall structure for processors.

I am temporarily blocking this until we figure out what is the right way forward and the right processor to put this in.

@bogdandrutu if you had discussed this earlier and had a resolution please comment.

**Comment by jrcamp** [comment]
@tigrannajaryan @bogdandrutu Given that it's going to probably be a while until a new metrics processor can be designed and implemented maybe we should take relatively trivial improvements to this processor in the meantime? (Obviously 'trivial' will then be debatable but I would say this falls into that bucket)

**Comment by bogdandrutu** [comment]
@punya I am ok to merge this if you sign to do the work of transforming this processor to pdata :)

**Comment by punya** [comment]
@bogdandrutu that helps, let me discuss with the team and get back on that in the next couple of days.

**Comment by punya** [comment]
@bogdandrutu @quentinmit this is still on my radar, I had to take off for a bit because of family issues.

**Comment by punya** [comment]
@bogdandrutu the Google team is signing up to do the work to convert this from OC to OTel data types. Please merge this PR in the meantime.

**Comment by qingling128** [comment]
@bogdandrutu - Anything else we need to address before this PR could be merged?

**Comment by qingling128** [comment]
Ah, looks like it needs to be rebased. @quentinmit - Could you help with that?

**Comment by quentinmit** [comment]
I just force-pushed this PR with a rebased patchset on top of current `main`.

**Comment by qingling128** [comment]
Hi @bogdandrutu @tigrannajaryan - Anything else we need to do before this could be merged? All the tests seem to pass. Previous comment / feedback have been addressed. And Google is signing up to do the work to convert this from OC to OTel data types.

CC @punya for visibility

**Comment by sophieyfang** [comment]
This is currently blocking to adjust our three metrics that we want to go GA with. Now those are in 1/1000th.

agent.googleapis.com/disk/weighted_io_time
agent.googleapis.com/disk/{io,operation}_time

**Comment by tigrannajaryan** [comment]
> @punya I am ok to merge this if you sign to do the work of transforming this processor to pdata :)

> @bogdandrutu the Google team is signing up to do the work to convert this from OC to OTel data types. Please merge this PR in the meantime.

@bogdandrutu do you have any other concerns or good to merge?

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**
